src/coprocessor/checksum.rs

```diff
@@ -78,7 +92,7 @@ impl<S: Snapshot> RequestHandler for ChecksumContext<S> {
     }
 }
 
-fn checksum_crc64_xor(checksum: u64, k: &[u8], v: &[u8]) -> u64 {
+pub fn checksum_crc64_xor(checksum: u64, k: &[u8], v: &[u8]) -> u64 {
```

Comment:
I suggest we change this function's signature to

```suggestion
pub fn checksum_crc64_xor(checksum: u64, k_prefix: &[u8], k_suffix: &[u8], v: &[u8]) -> u64 {
```

so that we don't need to allocate a new `Vec` during undo-rewrite-prefix.

Reply:
good idea!